runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -822,40 +865,47 @@ int chpl_task_createCommTask(chpl_fn_p fn,
                           NULL, comm_task_wrapper, &wrapper_info);
 }
 
-void chpl_task_addToTaskList(chpl_fn_int_t     fid,
-                             void             *arg,
-                             c_sublocid_t      subloc,
-                             void            **task_list,
-                             int32_t           task_list_locale,
-                             chpl_bool         is_begin_stmt,
-                             int               lineno,
-                             int32_t           filename)
+void chpl_task_addToTaskList(chpl_fn_int_t       fid,
+                             chpl_task_bundle_t *arg,
+                             size_t              arg_size,
+                             c_sublocid_t        subloc,
+                             void              **task_list,
+                             int32_t             task_list_locale,
+                             chpl_bool           is_begin_stmt,
+                             int                 lineno,
+                             int32_t             filename)
 {
     chpl_bool serial_state = chpl_task_getSerial();
+    chpl_fn_p requested_fn = chpl_ftable[fid];
 
     assert(subloc != c_sublocid_none);
 
     PROFILE_INCR(profile_task_addToTaskList,1);
 
     if (serial_state) {
         // call the function directly.
-        (chpl_ftable[fid])(arg);
+        requested_fn(arg);
     } else {
-        chpl_qthread_wrapper_args_t wrapper_args =
-            {chpl_ftable[fid], arg, NULL, false,
-             PRV_DATA_IMPL_VAL(filename, lineno, chpl_nullTaskID, false,
-                               subloc, serial_state) };
-
-        wrap_callbacks(chpl_task_cb_event_kind_create,
-                       &wrapper_args.chpl_data);
+        arg->serial_state      = false;
+        arg->countRunning      = false;
+        arg->is_executeOn      = false;
+        arg->requestedSubloc   = subloc;
+        arg->requested_fn      = requested_fn;
+        arg->task_prv.lineno   = lineno;
+        arg->task_prv.filename = filename;
+
+        // wrap_callbacks assigns the task id on create?
+        // but that seems nasty. Also, we no longer need
+        // to form a chpl_task_prvDataImpl_t at this point.
+        // Can we call the event inside the task once it runs?
+        // wrap_callbacks(chpl_task_cb_event_kind_create,
```

Comment:
I believe it does, in order to allow chplvis to figure out the parent/child relationships between tasks.  But check with Phil to be sure.